### PR TITLE
rootless: support `--exec-opt native.cgroupdriver=systemd`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**

Support rootless cgroup.

Requires cgroup v2 host with crun.
Tested with Ubuntu 19.10 (kernel 5.3, systemd 242), crun v0.12.1.

**- How I did it**

Using systemd.

**- How to verify it**
* Install Ubuntu 19.10
* Install containerd master
* Install crun v0.12.1
* Add `systemd.unified_cgroup_hierarchy=1` to the kernel cmdline
* (Optional) To enable the CPU controller:
```console
$ cat > /etc/systemd/system/user@.service.d/foo.conf << EOF
[Service]
# default: Delegate=pids memory
Delegate=pids memory cpu
EOF
```

Start a daemon:

```console
$ dockerd-rootless.sh \
  --experimental \
  --exec-opt native.cgroupdriver=systemd \
  --add-runtime crun=crun \
  --default-runtime=crun
```

Verify:
```console
$ export DOCKER_HOST=unix:///run/user/1001/docker.sock

$ docker  run -d --memory 32M --cpus 0.5 nginx:alpine
71500664473ea71aff8360ba31db7eaee4b11fba19c4fdd467ceaa49c2fef7e0


$ cat /sys/fs/cgroup/user.slice/user-1001.slice/user@1001.service/user.slice/docker-71500664473ea71aff8360ba31db7eaee4b11fba19c4fdd467ceaa49c2fef7e0.scope/memory.max 
33554432

$ cat /sys/fs/cgroup/user.slice/user-1001.slice/user@1001.service/user.slice/docker-71500664473ea71aff8360ba31db7eaee4b11fba19c4fdd467ceaa49c2fef7e0.scope/cpu.max 
50000 100000
```

**- Description for the changelog**

rootless: support `--exec-opt native.cgroupdriver=systemd`

**- A picture of a cute animal (not mandatory but encouraged)**
:penguin:

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
cc @giuseppe FYI for Podman compatibility

File: rootless/specconv/specconv_linux.go [hunk 1]
```diff
@@ -26,10 +27,19 @@ func getCurrentOOMScoreAdj() int {
...
+	if noCgroup {
+		// Remove cgroup settings.
+		spec.Linux.Resources = nil
+		spec.Linux.CgroupsPath = ""
+	} else {
+		if spec.Linux.Resources != nil {
+			// Remove devices: https://github.com/containers/crun/issues/255
+			spec.Linux.Resources.Devices = nil
+			// Remove blockIO as a workaround of https://github.com/containers/crun/issues/254
+			spec.Linux.Resources.BlockIO = nil
```
Comment (AkihiroSuda) [inline rootless/specconv/specconv_linux.go]: Maybe we should parse cgroup.subtree_control (via systemd API, if exists?) and do this conditionally.

**Comment by AkihiroSuda** [comment]
@tonistiigi PTAL?

**Review by tonistiigi — COMMENTED** [review]
SGTM

**Comment by AkihiroSuda** [comment]
@thaJeztah PTAL?

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by AkihiroSuda** [comment]
@tonistiigi LGTY?

**Review by tonistiigi — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
PR for runc: https://github.com/opencontainers/runc/pull/2281

**Pull request merged**